gui/src/Event/Event.cpp

```diff
@@ -192,3 +192,8 @@ void Gui::Event::decreaseTimeUnit()
     if (_gameData.get()->getTimeUnitFromServer() == GameData::TimeUnitState::NONE)
         _gameData.get()->setTimeUnitFromServer(GameData::TimeUnitState::DECREASE);
 }
+
+void Gui::Event::setPlayerVision()
+{
+    _render.get()->setPlayerVision(!_render.get()->getPlayerVision());
```

Comment:
You can't do this ? (With an arrow)

```suggestion
    _render->get()->setPlayerVision(!_render.get()->getPlayerVision());
```

Reply:
Because here, `_render` is the variable, to get the pointer itself, you do .get(), then (as already written), you can use the arrow to call you methods, etc...